entc/gen/template/meta.tmpl

```diff
@@ -134,6 +134,15 @@ var (
 					return fmt.Errorf("{{ $.Package }}: invalid enum value for {{ $f.Name }} field: %q", {{ $arg }})
 			}
 		}
+
+        // {{$enum}}All returns all values of {{$enum}} enum
+        func {{$enum}}All() []{{$enum}} {
+                return []{{$enum}}{
+                    {{- range $_, $e := $f.Enums}}
+                        {{ $f.EnumName $e }},
+                    {{- end}}
+                }
+        }
 	{{ end }}
 {{ end }}
 
```

Comment:
If we decide to go with this, I think it should be an array of values and not a function. 
If we won't, you can always add this tiny logic to your codegen by using templates (the one below:  `"meta/additional"`). 

Thanks for your contribution!

Reply:
So, based on your advice codegen should look like:

```
var StateAll = [...]State{
    StateOn,
    StateOff,
}
```
?

The reason is why this was wrapped into function is because slices and arrays are mutable, so them can accidentally be changed.

Also not sure about naming. Is it okay to just suffix a enum name with `All`?

Reply:
> The reason is why this was wrapped into function is because slices and arrays are mutable, so them can accidentally be changed.

I agree. 


> You can always add this tiny logic to your codegen by using templates

About my previous comment -  I don't know what @alexsn thinks about it, but it looks to me as something that can be solved easily with an external template. If you are not sure how to use it, here's an example - [entc/integration/template](https://github.com/facebookincubator/ent/blob/master/entc/integration/template/ent/generate.go).

Reply:
@a8m okay, i will close this, better to keep this for users implementation.

Reply:
Thanks for the understanding @danf0rth 
I usually try to push stuff to the base of the codegen after I found them widely used.

I can share with you that @alexsn and I have discussions about creating an `ent-integration` repo for `ent`, so we can share all the other stuff we do that no part of the base codegen. For example, GraphQL Node API, [Relay pagination](https://facebook.github.io/relay/graphql/connections.htm), etc.  

This one, can reside there as well.